### PR TITLE
fix marketplace icon parse issue

### PR DESCRIPTION
**Pull request by SLdragon**

### CODE REVIEW

File: src/marketplace/localserver.ts [hunk 1]
```diff
@@ -89,20 +89,22 @@ export class LocalServer {
...
-                        item.iotEdgeMetadataUrl = metaData.uri;
-                    } else {
-                        item.iotEdgeMetadataUrl = null;
-                    }
+                    if (item.plans && item.plans[0] && item.plans[0].artifacts) {
+                        const metaData = item.plans[0].artifacts.find((artifact) => artifact.name === "iot-edge-metadata.json");
+                        if (metaData !== null) {
+                            item.iotEdgeMetadataUrl = metaData.uri;
+                        } else {
+                            item.iotEdgeMetadataUrl = null;
```
Comment (adashen) [inline src/marketplace/localserver.ts]: what will happen if matadata url is null? although not related to the hot fix. This maybe an issue we need take care.

File: src/marketplace/localserver.ts [hunk 2]
```diff
@@ -89,20 +89,22 @@ export class LocalServer {
...
+                        }
 
-                    const image = item.images[0].items.find((img) => img.id === "small");
-                    if (image !== null) {
-                        item.icon = image.uri;
-                    } else {
-                        item.icon = null;
+                        const iconFileUris = item.iconFileUris;
```
Comment (formulahendry) [inline src/marketplace/localserver.ts]: This code block should not be inside `if (item.plans && item.plans[0] && item.plans[0].artifacts)` block

Comment (SLdragon) [reply src/marketplace/localserver.ts]: If we cannot parse plans, do we need to skip this module?

Comment (adashen) [reply src/marketplace/localserver.ts]: If we could not get metadata, the current code will get error when user click import right? If so, I think we may not put the invalid modules in to item list

Comment (SLdragon) [reply src/marketplace/localserver.ts]: Ok, if we cannot find the meta url, then server side will not send this module to front end

**Pull request merged**